src/scripts/main.js

```diff
@@ -1,3 +1,206 @@
+        name="age"
+        type="number"
+        data-qa="age"
+      />
+    </label>
+    <label>
+      Salary:
+      <input
+        name="salary"
+        type="number"
+        data-qa="salary"
+      />
+    </label>
+    <button
+      type="button"
+      id="save-button"
+    >
+      Save to table
+    </button>
+  </form>
+  `;
+
+table.insertAdjacentHTML('afterend', formHtml);
+
+// 4. Show notification if form data is invalid
+function pushNotification(message, type) {
+  const notification = document.createElement('div');
+  const titleSpan = document.createElement('span');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+
+  titleSpan.className = 'title';
+  titleSpan.textContent = type.toUpperCase();
+
+  notification.appendChild(titleSpan);
+  notification.innerHTML += ` <span>${message}</span>`;
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+}
+
+function validateFormData(formData) {
+  if (formData.name.length < 4) {
+    pushNotification('Name must be at least 4 characters long.', 'error');
+
+    return false;
+  }
+
+  if (formData.age < 18 || formData.age > 90) {
+    pushNotification('Age must be between 18 and 90.', 'error');
+
+    return false;
+  }
+
+  if (!formData.position || !formData.office || !formData.salary) {
+    pushNotification('All fields are required.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+const form = document.querySelector('.new-employee-form');
+
+document.getElementById('save-button').addEventListener('click', () => {
+  const formData = [...form.elements].reduce((acc, input) => {
+    if (input.name) {
+      acc[input.name] = input.value;
+    }
+
+    return acc;
+  }, {});
+
+  if (validateFormData(formData)) {
+    const newRow = document.createElement('tr');
+
+    newRow.innerHTML = `
+      <td>${formData.name}</td>
+      <td>${formData.position}</td>
+      <td>${formData.office}</td>
+      <td>${formData.age}</td>
+      <td>$${parseInt(formData.salary).toLocaleString('en-US')}</td>`;
+
+    table.querySelector('tbody').appendChild(newRow);
+    pushNotification('Employee added successfully!', 'success');
+    form.reset();
+  }
+});
+
+// 5. Implement editing of table cells by double-clicking on it. (optional)
+table.querySelectorAll('tbody td').forEach((cell) => {
+  cell.addEventListener('dblclick', () => {
+    const initialValue = cell.innerText;
+    const input = document.createElement('input');
+
+    input.className = 'cell-input';
+    input.value = initialValue;
+    cell.innerText = '';
+    cell.appendChild(input);
+    input.focus();
+
+    input.addEventListener(
+      'blur',
+      () => (cell.innerText = input.value || initialValue),
+    );
+
+    input.addEventListener('keypress', (e) => {
+      if (e.key === 'Enter') {
+        cell.innerText = input.value || initialValue;
+      }
+    });
+  });
+});
```

Comment:
The event listener on the 'dblclick' event creates a new input element every time a cell is double clicked. This could lead to performance issues if the user double clicks many times. It would be better to create the input element once and reuse it.